src/sc_keeper/crawler_extend.py

```
@@ -0,0 +1,242 @@
+        else:
+            tier_dicts = price_tiers_json
+
+        if not tier_dicts or not isinstance(tier_dicts, list):
+            return []
+
+        price_tiers = []
+        for tier in tier_dicts:
+            if tier.get("upper") == "Infinity":
+                tier["upper"] = float("inf")
+            price_tiers.append(PriceTier(**tier))
+        return price_tiers
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)